bigquery/src/main/scala/com/yahoo/maha/executor/bigquery/BigqueryQueryExecutor.scala

```diff
@@ -0,0 +1,227 @@
+// Copyright 2017, Yahoo Holdings Inc.
+// Licensed under the terms of the Apache License 2.0. Please see LICENSE file in project root for terms.
+package com.yahoo.maha.executor.bigquery
+
+import com.fasterxml.jackson.annotation.JsonProperty
+import com.fasterxml.jackson.databind.{DeserializationFeature, ObjectMapper}
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory
+import com.fasterxml.jackson.module.scala.{DefaultScalaModule, ScalaObjectMapper}
+import com.github.vertical_blank.sqlformatter.scala.{FormatConfig, SqlDialect, SqlFormatter}
+import com.google.api.client.http.apache.v2.ApacheHttpTransport
+import com.google.auth.http.HttpTransportFactory
+import com.google.auth.oauth2.ServiceAccountCredentials
+import com.google.cloud.ServiceOptions
+import com.google.cloud.bigquery.BigQuery
+import com.google.cloud.bigquery.BigQueryOptions
+import com.google.cloud.bigquery.JobId
+import com.google.cloud.bigquery.JobInfo
+import com.google.cloud.bigquery.QueryJobConfiguration
+import com.google.cloud.http.HttpTransportOptions
+import com.yahoo.maha.core._
+import com.yahoo.maha.core.query._
+import com.yahoo.maha.core.request.{Parameter, RequestIdValue}
+import grizzled.slf4j.Logging
+import java.io.FileInputStream
+import java.util.UUID
+import org.apache.http.HttpHost
+import org.apache.http.auth.{AuthScope, UsernamePasswordCredentials}
+import org.apache.http.impl.client.{
+  BasicCredentialsProvider,
+  DefaultHttpRequestRetryHandler,
+  ProxyAuthenticationStrategy
+}
+import org.apache.http.impl.conn.DefaultProxyRoutePlanner
+import scala.util.Try
+
+private case class ProxyCredentials(
+  @JsonProperty("proxy_user") userName: String,
+  @JsonProperty("proxy_pass") password: String
+)
+
+case class BigqueryQueryExecutorConfig(
+  gcpCredentialsFilePath: String,
+  gcpProjectId: String,
+  enableProxy: Boolean,
+  proxyCredentialsFilePath: Option[String],
+  proxyHost: Option[String],
+  proxyPort: Option[String],
+  retries: Int
+) extends Logging {
+
+  private val mapper = {
+    val om = new ObjectMapper(new YAMLFactory()) with ScalaObjectMapper
+    om.registerModule(DefaultScalaModule)
+    om.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
+    om
+  }
+
+  def buildBigqueryClient(): BigQuery = {
+    val builder = BigQueryOptions
```

Comment:
You probably want to set the Connect and Read timeouts here and pass them in as config.

Reply:
See HttpTransportOptions

Reply:
Added these configs